crates/pathfinder/src/storage/schema/revision_0010.rs

```diff
@@ -299,6 +309,73 @@ mod tests {
                     Ok(())
                 }
             }
+
+            #[derive(Clone, Debug, PartialEq)]
+            pub struct Transaction {
+                pub calldata: Option<Vec<CallParam>>,
+                pub class_hash: Option<ClassHash>,
+                pub constructor_calldata: Option<Vec<ConstructorParam>>,
+                pub contract_address: Option<ContractAddress>,
+                pub contract_address_salt: Option<ContractAddressSalt>,
+                pub entry_point_type: Option<EntryPointType>,
+                pub entry_point_selector: Option<EntryPoint>,
+                pub max_fee: Option<Fee>,
+                pub nonce: Option<TransactionNonce>,
+                pub sender_address: Option<ContractAddress>,
+                pub signature: Option<Vec<TransactionSignatureElem>>,
+                pub transaction_hash: StarknetTransactionHash,
+                pub r#type: Type,
+                pub version: Option<TransactionVersion>,
+            }
+
+            pub struct StarknetEventsTable {}
+            impl StarknetEventsTable {
+                pub fn insert_events(
+                    connection: &Connection,
+                    block_number: StarknetBlockNumber,
+                    transaction: &Transaction,
+                    events: &[transaction::Event],
+                ) -> anyhow::Result<()> {
+                    if transaction.contract_address.is_none() && !events.is_empty() {
+                        anyhow::bail!("Declare transactions cannot emit events");
```

Comment:
We could include some unique identifier info here, like here it might be `{block_number}:{transaction.txn_hash}`, otherwise this will be very difficult to debug.